### PR TITLE
[release-4.9] OCPBUGS-15865: Dockerfile: build both RHEL7 and RHEL8 shims

### PR DESCRIPTION
**Pull request by jcaamano**
Since the shim (ovn-k8s-cni-overlay) gets copied to the host OS and executed in the host mount namespace by CRIO/Multus it needs to be runtime compatible with the host OS. Running a RHEL8-built shim on a RHEL7 system doesn't work due to different shared library dependencies between the two OS versions.

This wasn't a problem before because CGO_ENABLED=0 which essentially statically linked everything into the binary. But since we actually need CGO_ENABLED=1 (which ART forces on "official" builds anyway) to ensure we use OpenSSL's crypto for FIPS compliance, we run into the OS version problem with our binaries since they are really always built with CGO_ENABLED=1 anyway.

So... build two separate versions of ovn-kubernetes in different layers, and copy the RHEL7 shim into a special location where our container startup scripts can find it.

### CODE REVIEW

**Comment by jcaamano** [comment]
@dcbw PTAL

File: Dockerfile [hunk 1]
```diff
@@ -9,15 +9,19 @@
...
 #
 
+# Build RHEL-8 binaries
 FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.9 AS builder
-
 WORKDIR /go/src/github.com/openshift/ovn-kubernetes
 COPY . .
-
-# build the binaries
 RUN cd go-controller; CGO_ENABLED=0 make
```
Comment (dcbw) [inline Dockerfile:16]: @jcaamano since we're building with CGO_ENABLED=1 anyway, should we just change it here?

**Comment by dcbw** [comment]
@jcaamano did you mean to leave in the contrib/kind.yaml bits?

**Comment by jcaamano** [comment]
> @jcaamano did you mean to leave in the contrib/kind.yaml bits?

no :facepalm:

**Pull request merged**